ctapipe/instrument/subarray.py

```diff
@@ -392,17 +398,17 @@ def to_hdf(self, output_path, overwrite=False):
             serialize_meta=serialize_meta,
             overwrite=overwrite,
         )
-        for camera in self.camera_types:
+        for i, camera in enumerate(self.camera_types):
             camera.geometry.to_table().write(
                 output_path,
-                path=f"/configuration/instrument/telescope/camera/geometry_{camera}",
+                path=f"/configuration/instrument/telescope/camera/geometry_{i}",
                 append=True,
```

Comment:
probably best to give a format spec here like `{i:02d}` just in case there are more than 10 geometries.

Reply:
Is it really important that in the improbable case of more than 10 telescopes the entries are not natural sorted?

Reply:
Using a leading zero breaks backwards compatibility or would make providing backwards compatibility  harder, since `:02d` is not a valid format string for `str` values used before.